WordPress/Classes/Models/Blog+Lookup.swift

```diff
@@ -58,4 +69,83 @@ public extension Blog {
 
         return service.findBlog(withXmlrpc: xmlrpc, andUsername: username)
     }
+
+    @objc(countInContext:)
+    static func count(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().count(in: context)
+    }
+
+    @objc(wpComBlogCountInContext:)
+    static func wpComBlogCount(in context: NSManagedObjectContext) -> Int {
+        BlogQuery().hostedByWPCom(true).count(in: context)
+    }
+
+    @objc(selfHostedInContext:)
+    static func selfHosted(in context: NSManagedObjectContext) -> [Blog] {
+        (try? BlogQuery().hostedByWPCom(false).blogs(in: context)) ?? []
+    }
+}
+
+struct BlogQuery {
```

Comment:
I'm a fan of using nested types to show the connection between, in this case, a `Blog` and it's query object.

Would you consider using

```swift
extension Blog {

    struct Query {
```

I acknowledge this might make typing the code a big harder, because `BlogQuery` is probably faster than `Blog.Query`, but I think that having a clearer (IMHO) relationship is worth it.

☝️  This doesn't have to be addressed in this PR, if at all. It's a mechanical change that fit nicely in a standalone one. ☝️